### DISCUSSION TITLE
Deeplinks support for Android through Skip

### DISCUSSION THREAD

**Discussion by hepspl** [skiptools/skip#164] [Q&A]
Hi,
I tried to implement deeplinks (custom url scheme) for Android app but I couldn't find a way to do it using Skip tools.

This is what is needed to be on Kotlin and Jetpack Compose side:
`  composable(`
`        "app?path={path}",`
`        deepLinks = listOf(navDeepLink {`
`            uriPattern = "customScheme://{path}"`
`        }),`
`    ) { deeplink ->`
`        val path = deeplink.arguments?.getString("path")`
`        Text("Requested path = /$path")`
`    }`

I tried to follow https://skip.tools/docs/modules/skip-ui/#composeview but I don't see the way how to apply this.

Is there a chance for accomplishing this?
Any hints appreciated 🙏🏻

**Comment by aabewhite** [comment]
We haven't implemented deep links yet, and I'm having trouble thinking of a good way for you to do it yourself. The "composable(...) { ... }" clause you cite above needs to be in a NavHost, which means it sort of has to take over the top-level routing of your app entirely. This is something we'll need to experiment with internally before we can recommend an approach.

What is your level of need and timeline for deep link support?

**Comment by hepspl** [comment]
Thanks for the answer.
I plan to use it for implementing OAuth2 login in the app.
In iOS project I used AuthenticationServices with ASWebAuthenticationSession which is obviously not available on Android via Skip.
So I decided to do it on my own by opening login url in webbrowser and then reacting to post-login redirection on custom url scheme where auth code is passed.

So that's super crucial actually but I am not really in a hurry (I have 6 month for it).

Maybe you have a hint how such login could be implemented right now using different approach?
I guess OAuth2 login is a pretty popular thing so it will be definitely needed for many Skip adopters...
